### ISSUE TITLE
[LSP] Configuration priority when specified in the editor

### ISSUE THREAD

**Issue by rchl** [astral-sh/ruff#23075]
### Summary

The behavior of `configuration` and `configurationPreference` settings is either buggy or I don't understand it.

## Reproduce

Consider this case:
1. Ruff configuration exists at `~/abc.toml` (intentionally not named `ruff.toml` so that it's not automatically picked up) with following contents:
```toml
[format]
quote-style = "double"
```
3. Editor (let's assume VSCode) configuration is set to:
```json
{
    "ruff.configuration": "~/abc.toml",
    "ruff.configurationPreference": "filesystemFirst",
}
```
4. A `main.py` file exists with following contents:
```py
string = 'string'
```
5. `Format document` is triggered in the editor

## What happens

The editor configuration is not respected.
Formatting doesn't change single quotes to double quotes.

## What should happen

Since there is no file system configuration anywhere in the file system hierarchy, I would expect that it falls back to the `abc.toml` configuration specified in the editor.

Why is that not happening?

(This issue was reported for Sublime Text initially but after looking into it, I see the same behavior in VSCode so it doesn't appear to be editor specific.)

### Version

ruff 0.15.0

**Comment by rchl** [comment]
Actually I had another configuration at `~/.config/ruff/ruff.toml` that was messing with my testing. I think this works fine actually.
